addons/crm/models/crm_lead.py

```diff
@@ -1434,11 +1503,146 @@ def _update_automated_probabilities(self):
+                lambda lead: fields.Date.to_date(pls_start_date) <= fields.Date.to_date(lead.create_date))
+            if not pls_leads:
+                return {}, {}
+
+        # Extract target leads values
+        team_ids = self.env['crm.team'].with_context(active_test=False).search([]).ids + [0]  # If team_id is unset, consider it as team 0
+        if rebuild:  # rebuild is ok
+            domain = [
+                '&',
+                    ('create_date', '>=', pls_start_date),
+                    '|',
+                        ('probability', '=', 100),
+                        '&',
+                            ('probability', '=', 0), ('active', '=', False)
+              ]
+        else:  # increment
+            domain = [('id', 'in', pls_leads.ids)]
+        leads_values_dict = pls_leads._pls_get_lead_pls_values(domain=domain)
+
+        # split leads values by team_id
+        # get current frequencies related to the target leads
+        leads_frequency_values_by_team = dict((team_id, []) for team_id in team_ids)
+        leads_pls_fields = set()  # ensure to keep each field unique (can have multiple tag_id leads_values_dict)
+        for lead_id, values in leads_values_dict.items():
+            team_id = values.get('team_id', 0)  # If team_id is unset, consider it as team 0
+            lead_frequency_values = {'count': 1}
+            for field, value in values['values']:
+                if field != "probability":  # was added to lead values in batch mode to know won/lost state, but is not a pls fields.
+                    leads_pls_fields.add(field)
+                else:  # extract lead probability - needed to increment tag_id frequency. (proba always before tag_id)
+                    lead_probability = value
+                if field == 'tag_id':  # handle tag_id separatelly (as in One Shot rebuild mode)
+                    leads_frequency_values_by_team[team_id].append({field: value, 'count': 1, 'probability': lead_probability})
+                else:
+                    lead_frequency_values[field] = value
+            leads_frequency_values_by_team[team_id].append(lead_frequency_values)
+        leads_pls_fields = list(leads_pls_fields)
+
+        # get new frequencies
+        new_frequencies_by_team = {}
+        for team_id in team_ids:
+            # prepare fields and tag values for leads by team
+            new_frequencies_by_team[team_id] = self._pls_prepare_frequencies(
+                leads_frequency_values_by_team[team_id], leads_pls_fields, target_state=target_state)
+
+        # get existing frequencies
+        existing_frequencies_by_team = {}
+        if not rebuild:  # there is no existing frequency in rebuild mode as they were all deleted.
+            # read all fields to get everything in memory in one query (instead of having query + prefetch)
+            existing_frequencies = self.env['crm.lead.scoring.frequency'].search_read(
+                [('variable', 'in', leads_pls_fields)])
```

Comment:
It seems to me not very efficient to load all the frequency of all the team when you only care about lead from few of the team. all the data of other team are loaded for nothing